vbeam/core/probe_geometry.py

```diff
@@ -0,0 +1,109 @@
+"""A datastructure for representing a transducer probe excluding 
+the position, orientation, etc.
+"""
+
+from typing import Callable, Optional
+from vbeam.fastmath import numpy as np
+from vbeam.fastmath.traceable import traceable_dataclass
+from vbeam.util.geometry.v2 import distance
+
+identity_fn = lambda x: x  # Just return value as-is
+
+
+@traceable_dataclass(("ROC", "rx_aperture_length_s", "tx_aperture_length_s"))
+class ProbeGeometry:
+    """A container for probe parameters.
+    The aperture lengths are arc lenghts of the probe surface in azimuth and elevation. 
+    """
+    ROC: tuple[float, float] # (azimuth, elevation)
+    rx_aperture_length_s: tuple[float, float] = None # (width, height)
+    tx_aperture_length_s: tuple[float, float] = None # (width, height)
+    def __postinit__(self):           
+        if type(self.ROC) is not tuple or len(self.ROC) != 2:
+            self.ROC = (self.ROC, self.ROC)
+    def __getitem__(self, *args) -> "ProbeGeometry":
+        _maybe_getitem = lambda attr: attr.__getitem__(*args) if attr is not None else None
+        return self.ProbeGeometry(
+            _maybe_getitem(self.ROC),
+            _maybe_getitem(self.rx_aperture_length_s),
+            _maybe_getitem(self.tx_aperture_length_s),
+        )
+
+    @property
+    def curvature_center(self):
+        return (np.array([0.0,0.0,-self.ROC[0]]), np.array([0.0,0.0,-self.ROC[1]]) ) 
+    
+    @property
+    def get_sender_normal(self):
+        vector = self.sender_position - self.curvature_center
```

Comment:
In general, I think it's a good idea to not set attributes on an object from outside of the object/class. I was a bit confused about where `self.sender_position` came from the first time I read this code. This property would be better as a method that takes the sender_position as an argument.

Is there another reason for setting `sender_position` on the probe from outside the probe object that I am missing?